Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -1555,9 +1562,14 @@ final class StatusTableViewController: LoopChartsTableViewController {
         }
 
         switch action {
-        case .presentViewController(let vc):
+        case .presentPumpViewController(let vc):
+            var completionNotifyingVC = vc
+            completionNotifyingVC.completionDelegate = self
+            self.present(completionNotifyingVC, animated: true, completion: nil)
+        case .presentCGMViewController(let vc):
```

Comment:
Rather than splitting these actions into pump/CGM (which seems weird to me: what happens when a PumpManager returns .presentCGMViewController?), perhaps the device manager can just return a VC that may or may not conform to a GlucoseUnitObserver protocol. If it does conform to the protocol, then you add the observer.